### PR TITLE
Admin endpoints support "client_credentials" tokens again

### PR DESCRIPTION
**Pull request by paulineribeyre**
Fix bug introduced in #1234 when using client_credentials tokens to hit admin endpoints:
```
Traceback: Traceback (most recent call last):
  File "/fence/.venv/lib/python3.9/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/fence/.venv/lib/python3.9/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/fence/fence/authz/auth.py", line 91, in wrapper
    return f(*f_args, **f_kwargs)
  File "/fence/fence/resources/audit/utils.py", line 128, in wrapper
    create_log_for_request(flask.request)
  File "/fence/fence/resources/audit/utils.py", line 103, in create_log_for_request
    username = get_user_from_claims(claims).username
  File "/fence/fence/auth.py", line 306, in get_user_from_claims
    .filter(User.id == claims["sub"])
KeyError: 'sub'
```

### Bug Fixes
- Admin endpoints support "client_credentials" tokens again

### Improvements
- Request logs now include the client ID, if any

### CODE REVIEW

File: fence/resources/audit/utils.py [hunk 1]
```diff
@@ -100,7 +100,8 @@ def create_log_for_request(request: Request):
     record a log entry.
     """
     claims = validate_jwt()
-    username = get_user_from_claims(claims).username
+    username = get_user_from_claims(claims).username if "sub" in claims else "None"
+    client_id = claims.get("azp", "None") or "None"
```
Comment (pieterlukasse) [inline fence/resources/audit/utils.py:104]: should we raise an error if both username and client end up being None?

**Review by pieterlukasse — COMMENTED** [review]
looks good to me. Just a few minor questions / comments.

File: fence/resources/audit/utils.py (hunk 1 above)
Comment (paulineribeyre) [reply fence/resources/audit/utils.py:104]: i don't think so, this is just the logging, the actual functional code would catch this (arborist, etc)

File: tests/admin/test_admin_users_endpoints.py [hunk 2]
```diff
@@ -330,7 +330,7 @@ def test_post_user(client, admin_user, encoded_admin_jwt, db_session, log_captur
     assert len(log_capture) >= 1
     # Now check for the specific message:
     messages = [f"{r.levelname} - {r.getMessage()}" for r in log_capture]
-    expected_log_message = "INFO - Incoming request: user=admin_user, method=POST, endpoint=/admin/user, request_url=/admin/user"
+    expected_log_message = "INFO - Incoming request: user=admin_user, client=None, method=POST, endpoint=/admin/user, request_url=/admin/user"
```
Comment (pieterlukasse) [inline tests/admin/test_admin_users_endpoints.py:333]: maybe also add a test where user=None?

**Review by pieterlukasse — DISMISSED** [review]

**Review by pieterlukasse — APPROVED** [review]
